### PR TITLE
Rename edgedict -> configstore.

### PR DESCRIPTION
**Pull request by dgryski**
This includes deprecation notices for the types and methods in edgedict, as well as semgrep patterns to update to the new API.

### CODE REVIEW

File: _integration_tests/fixtures/configstore/fastly.toml [hunk 1]
```diff
@@ -23,6 +23,6 @@ service_id = ""
 
   [local_server.dictionaries]
 
-    [local_server.dictionaries.edge_dictionary]
-      file = "./_integration_tests/fixtures/edge_dictionary/edge_dictionary.json"
+    [local_server.dictionaries.configstore]
```
Comment (dgryski) [inline _integration_tests/fixtures/configstore/fastly.toml:26]: viceroy still calls them `dictionaries` I guess.

File: configstore/doc.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Copyright 2022 Fastly, Inc.
+
+// Package configstore provides a read-only interface to config stores. Refer
+// to https://developer.fastly.com/reference/api/dictionaries/ for more
```
Comment (dgryski) [inline configstore/doc.go:4]: No new page to link to for "config stores"?

Comment (joeshaw) [reply configstore/doc.go:4]: https://developer.fastly.com/reference/api/services/resources/config-store/, though this might change.

**Review by joeshaw — APPROVED** [review]
One doc link update below, but otherwise LGTM

File: semgrep/edgedict.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+rules:
```
Comment (joeshaw) [inline semgrep/edgedict.yml:1]: Might be good to include some docs or a makefile rule or something to show how to use these.  Or include them in CI.

Comment (dgryski) [reply semgrep/edgedict.yml:1]: Oh right.  I'll include a README explaining that these are.

**Comment by dgryski** [comment]
PTAL

**Comment by fgsch** [comment]
Instead of rename can we keep the edgedict around and add a deprecated notice to it?

**Comment by dgryski** [comment]
@fgsch Yes, that's exactly what this PR does.  It leaves a deprecated stub in `edgedict` that calls out to the `configstore` package as the base implementation, as well as adding deprecation notices to anything using `edgedict`.  It also uses a type alias for `edgedict.Dictionary` pointing to `configstore.Store`, and has a set of semgrep rules to handle the migration automatically (except for import rewriting).

**Review by JakeChampion — APPROVED** [review]

**Comment by fgsch** [comment]
@dgryski doh, I should have read this more carefully! Sorry for the noise.

**Pull request merged**